### PR TITLE
Prmdr 168 Back Channel Logout

### PR DESCRIPTION
**Pull request by thisusernameisnowtaken**

### CODE REVIEW

File: lambdas/handlers/back_channel_logout_handler.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        logger.info("decoding token")
+        oidc_service = OidcService()
+        decoded_token = oidc_service.validate_and_decode_token(token)
+        session_id = decoded_token["sid"]
+        remove_session_from_dynamo_db(session_id)
+
+    except ClientError as e:
+        logger.error(f"Error logging out user: {e}")
+        return ApiGatewayResponse(
+            400, """{ "error":"Internal error logging user out"}""", "POST"
```
Comment (NogaNHS) [inline lambdas/handlers/back_channel_logout_handler.py:40]: Shouldn't we return 500 here?

File: lambdas/handlers/back_channel_logout_handler.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        return ApiGatewayResponse(
+            400, """{ "error":"No sid field in decoded token"}""", "POST"
+        ).create_api_gateway_response()
+
+    return ApiGatewayResponse(200, "", "POST").create_api_gateway_response()
+
+
+def remove_session_from_dynamo_db(session_id):
+    logger.info(f"Session to be removed: {session_id}")
+    dynamodb_name = os.environ["AUTH_DYNAMODB_NAME"]
```
Comment (NogaNHS) [inline lambdas/handlers/back_channel_logout_handler.py:58]: This env var is missing from '@ensure_environment_variables'

**Review by SRAlexander — APPROVED** [review]

File: lambdas/handlers/back_channel_logout_handler.py (hunk 1 above)
Comment (AlexHerbertNHS) [reply lambdas/handlers/back_channel_logout_handler.py:40]: Done

File: lambdas/handlers/back_channel_logout_handler.py (hunk 2 above)
Comment (AlexHerbertNHS) [reply lambdas/handlers/back_channel_logout_handler.py:58]: Done

**Pull request merged**